pkgs/applications/networking/instant-messengers/element/keytar/default.nix

```diff
@@ -31,6 +31,17 @@ in stdenv.mkDerivation rec {
 
   doCheck = false;
 
+  postPatch = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+    pkg-config() { "''${PKG_CONFIG}" "$@"; }
+    export -f pkg-config
+  '';
+
+  # https://nodejs.org/api/os.html#osarch
+  npmFlagsArray = [ "--arch=${if stdenv.hostPlatform.parsed.cpu.arch == "i686" then "ia32"
+                             else if stdenv.hostPlatform.parsed.cpu.arch == "x86_64" then "x64"
+                             else if stdenv.hostPlatform.parsed.cpu.arch == "aarch64" then "arm64"
+                             else stdenv.hostPlatform.parsed.cpu.arch}" ];
```

Comment:
I have to admit I'm not too familiar with `lib/systems`, but e.g. `nix-instantiate -A pkgsCross.mipsel-linux-gnu.stdenv.hostPlatform.parsed.cpu --eval` doesn't have an arch set.

Reply:
Looks like this was the wrong attribute to begin with. I am surprised it worked at all.

Reply:
Make me wonder if it matters at all what we pass in this case.